EDU-826: Migrate legacy ECMAScript modules content into TypeScript dev guide (#2125)
temporalio/documentation: docs-src/typescript/how-to-import-an-ecmascript-module.md

---
id: how-to-import-an-ecmascript-module
title: How to import an ECMAScript module
sidebar_label: ECMAScript modules
description: See our Fetch ESM sample for the necessary configuration changes.
tags:
  - guide-context
---

> The JavaScript ecosystem is increasingly moving toward publishing ECMAScript modules (ESM) instead of CommonJS modules; for example, `node-fetch@3` is ESM, but `node-fetch@2` is CommonJS.

Reviewer: ```suggestion
The JavaScript ecosystem is quickly moving toward publishing ECMAScript modules (ESM) instead of CommonJS modules.
For example, `node-fetch@3` is ESM, but `node-fetch@2` is CommonJS.
```

Reviewer: Also, I would add a mention about what this means for updates. Like, keep an eye on this ecosystem change because it can affect code that uses these modules, something like that.

> If you want to import a pure ESM dependency, see our [Fetch ESM](https://github.com/temporalio/samples-typescript/tree/main/fetch-esm) sample for the necessary configuration changes:

Reviewer: ```suggestion
For more information on importing a pure ESM dependency, see our [Fetch ESM](https://github.com/temporalio/samples-typescript/tree/main/fetch-esm) sample for the necessary configuration changes:
```
